### PR TITLE
Initial class structure

### PR DESCRIPTION
**Pull request by lanehunter-dev**
* **Please check if the PR fulfills these requirements**
- [x] PR uses multiple, atomic commits
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] PR buddy has been requested to review this PR

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR adds in a view for the login screen, user dashboard, and adds class methods to filter trip data pulled from the API.

* **What is the current behavior?** (You can also link to an open issue here)
- Currently the page is set to auto login a user defined in the index.js.
- From there a user can view all of their trips and filter by past trips, current trips, upcoming trips, and pending trips.

* **What is the new behavior (if this is a feature change)?**
N/A

* **Other information**:
There are no tests for most of the methods in the classes, testing will come in the next PR.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,13 +1,49 @@
 <!DOCTYPE html>
 <html lang="en-US">
   <head>
-    <title>Webpack Starter Kit</title>
+    <link href="https://fonts.googleapis.com/css?family=Arvo:400,700|Roboto&display=swap" rel="stylesheet">
+    <title>Freudian Trip</title>
```
Comment (grimeyg) [inline src/index.html]: Like slip, nice 👌🏻

File: src/index.js [hunk 2]
```diff
@@ -1,13 +1,85 @@
...
+import Traveler from './traveler.js';
+import Agent from './agent.js'
+import './images/057-placeholder.svg'
+let currentUser;
+let destinationData;
+let tripData;
+
+autoLogin()
+
+$('.login-btn').click(attemptLogin);
```
Comment (grimeyg) [inline src/index.js]: This, and line 35 are very large blocks of code, you can possibly refactor

File: src/index.js [hunk 3]
```diff
@@ -1,13 +1,85 @@
...
+    domUpdates.displayCurrentTrips(currentUser)
+  } else if (value === 'upcoming') {
+    domUpdates.displayUpcomingTrips(currentUser)
+  } else if (value === 'pending') {
+    domUpdates.displayPendingTrips(currentUser)
+  }
+})
+
+
+async function attemptLogin() {
```
Comment (grimeyg) [inline src/index.js]: Nice job getting async working!

File: src/traveler.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  getTotalTripCostPerYear() {
+    let subTotal = this.trips.reduce((totalCost, trip) => {
+      let place = this.destinations.find(destination => destination.id === trip.destinationID);
+      totalCost += (place.estimatedLodgingCostPerDay * trip.duration * trip.travelers);
+      totalCost += (place.estimatedFlightCostPerPerson * trip.travelers);
+      return totalCost;
+    }, 0)
+    return subTotal;
+  }
+  getPastTrips() {
```
Comment (grimeyg) [inline src/traveler.js]: You got the dates working nice job!

File: src/css/base.scss [hunk 5]
```diff
@@ -1,6 +1,258 @@
...
+  border-radius: .3em;
+  padding: 1em;
+  margin-top: .3em;
+  margin-left: 10%;
+  width: 80%;
+  height: 3em;
+  border: none;
+}
+
+button:hover {
```
Comment (grimeyg) [inline src/css/base.scss]: Oof dat microinteraction

Comment (lanehunter-dev) [reply src/css/base.scss]: @khalidwilliams

**Pull request merged**